src/Components/Task.js

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useState } from 'react';
+
+const Task = ({ description, setTaskList, tasks, id, setTaskToBeEdited, setNewDescription }) => {
+    const [isStrikeThrough, setStrikeThrough] = useState(false);
+
+    const removeTask = (id) => {
+        return setTaskList(tasks.filter(task => task.id !== id))
+    };
+
+    useEffect(() => {
+        const modal = document.querySelector('.modal');
+        const overlay = document.querySelector('.overlay');
+        const editButtons = Array.from(document.querySelectorAll('.button__edit'));
+        
+        function showModal() {
+            modal.classList.remove('hidden');
+            overlay.classList.remove('hidden');
+        }
```

Comment:
We always declare functions inside the component and outside the `useEffect` hook. This is logically correct but not a good approach. Also try to use arrow functions as it is the latest standard.